### PR TITLE
Add waitFor utility

### PR DESCRIPTION
**Pull request by spkrka**

### CODE REVIEW

**Comment by spkrka** [comment]
@danielnorberg @rschildmeijer

File: src/test/java/com/spotify/futures/FuturesExtraTest.java [hunk 1]
```diff
@@ -571,4 +572,78 @@ public void testAsyncException6() throws Exception {
...
+
+    assertFalse(result.isDone());
+    other3.set("foo");
+    assertTrue(result.isDone());
+    assertEquals("hello", result.get());
+
+  }
+
+  @Test(expected = ExecutionException.class)
+  public void testWaitForNotImmediateWithFailure() throws Exception {
```
Comment (pettermahlen) [inline src/test/java/com/spotify/futures/FuturesExtraTest.java]: Wouldn't it be better semantics to have waitFor fail and report the exception if any of the waited for futures fail? It seems like you'd want to know if something fails. And if you don't, you can always use a fallback or something to ensure the other futures you're waiting for never fail.

File: README.md [hunk 2]
```diff
@@ -170,6 +170,20 @@ final ListenableFuture<B> future = getFuture();
...
 
+#### Waiting for multiple futures
+
+Sometimes you want to depend on a single future, but for timing reasons it's useful to
+wait for other futures to complete. You can use this method for that:
+```java
+final ListenableFuture<String> future = Futures.immediateFuture("hello");
+final ListenableFuture<String> other1 = Futures.immediateFuture("other1");
+final ListenableFuture<String> other2 = Futures.immediateFuture("other2");
+final ListenableFuture<String> result = FuturesExtra.waitFor(future, other1, other2);
```
Comment (pettermahlen) [inline README.md]: `waitFor` doesn't really convey what it does to me, but I think it's hard to come up with a better name. It's something like 'returnResultOfFirstWhenAllDone'. Maybe `waitAndReturnFirst` ?

File: src/test/java/com/spotify/futures/FuturesExtraTest.java (hunk 1 above)
Comment (spkrka) [inline src/test/java/com/spotify/futures/FuturesExtraTest.java]: For that usecase I would simply use FuturesExtra.syncTransformN or a transform with Futures.allAsList (which will fail if any waited fails)

This is only useful if you want to get some sort of ordering guarantees but don't want to be affected by other futures.

File: README.md (hunk 2 above)
Comment (spkrka) [inline README.md]: The naming is not super clear, but hopefully the javadoc of the function should make it more obvious what it does.

**Comment by caesar-ralf** [comment]
closing old PR

**Pull request closed without merging**